Python/Module6_Testing/Intro_to_Testing.md

```diff
@@ -0,0 +1,532 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.3.0
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+<!-- #raw raw_mimetype="text/restructuredtext" -->
+.. meta::
+   :description: Topic: Writing tests for your code, Difficulty: Easy, Category: Section
+   :keywords: test, automated, unit, assert  
+<!-- #endraw -->
+
+# Introduction to Testing
+
+This section will show us just how simple it is to write rudimentary tests. We need only recall some of Python's basic scoping rules and introduce ourselves to the `assert` statement to write a genuine test function. That being said, we will quickly encounter some important questions to ponder. How do we know that our tests work? And, how do we know that our tests are effective? These questions will drive us deeper into the world of testing. 
+
+Before we hit the ground running, let's take a moment to consider some motivations for testing out code.
+
+<!-- #region -->
+## Why Should We Write Tests?
+
+The fact of the matter is that it is intuitive for most people to test their code to some extent.
+After writing, say, a new function, it is only natural to contrive an input to feed it, and to check that the function returns the output that we expected.
+To the extent that one would want to see evidence that their code works, we need not motivate the importance of testing.
+
+Less obvious are the massive benefits that we stand to gain from formalizing this testing process.
+And by "formalizing", we mean taking the test scenarios that we were running our code through, and encapsulating them in their own functions that can be run from end-to-end.
```

Comment:
I'd use "automating" instead of "formalizing" - the latter has a lot of mathematical baggage, especially relating to software correctness!